ag_data/models.py

```diff
@@ -60,10 +60,22 @@ class AGSensor(models.Model):
     """Stores the information about sensors including name and type id.
     """
 
-    id = models.AutoField(primary_key=True)
+    uuid = models.UUIDField(primary_key=True, default=uuid.uuid4, editable=False)
+    serial = models.IntegerField(unique=True, default=1)
     name = models.CharField(max_length=1024, blank=True)
     type_id = models.ForeignKey(AGSensorType, null=False, on_delete=models.PROTECT)
 
+    def save(self, *args, **kwargs):
+        if self._state.adding:
```

Comment:
This PR contains two major changes: 1) Set primary key as uuid, 2) overriding `save` of AGSensor model. I guess the overriding of save function is the core of this PR.

Even without using uuid as a primary key, I guess this override could resolve the primary key issue. Because the problem is from where django isn't aware of the increased primary key. If so, we can stick with using primary key as integer but just overriding the save method. 

However I'm not 100% sure if it's right and I'm okay with the current implementation. So I won't request for change for this reason as mandatory. It's up to you.

Reply:
Yes indeed, I figured that it would still work if we only override `save` without changing the primary key. That being said, if we look at the overall system, having a primary key that is independent than the `serial` is beneficial. 

Essentially, this will have an internal and external key. the "internal" primary key (uuid) will be used in the system to do things including, associating with measurements. and the "external" one, which is the numerical `serial` is used for interfacing with the raspberry pi program. 

This will make the changes on the raspberry pi configuration independent from that in the client software. With the new design, the user can easily change a sensor serial on the UI, to suit their needs. And if the user indeed changes the serial, existing measurements are still linked to the same sensor, with future measurements with the new serial, but the same sensor in the database.